src/concurrent/channel.cr

```diff
@@ -269,4 +271,11 @@ class Channel::Unbuffered(T) < Channel(T)
   def full?
     @has_value || @receivers.empty?
   end
+
+  def close
+    super
+    if sender = @sender
```

Comment:
You should set `@sender = nil` here to prevent any possibility of `@sender` being enqueued twice.

Reply:
thanks!